src/printer.js

```diff
@@ -908,8 +907,10 @@ function genericPrintNoParens(path, options, print, args) {
       return nodeStr(n, options);
     case "UnaryExpression":
       parts.push(n.operator);
-      if ( n.argument && n.argument.type !== "UnaryExpression" && ( n.operator !== '+' && n.operator !== '-') ) {
-        parts.push(' ');
+
+      if (/[a-z]$/.test(n.operator)) parts.push(" ");
```

Comment:
I made this change to match the [configuration](https://github.com/Automattic/eslint-config-wpcalypso/blob/master/index.js#L81) of the `space-unary-ops` eslint rule. It adds space after textual unary ops (`typeof`, `delete`) and `!`, handles the `!!` special case, and doesn't add space after others -- `~`, `+`, `-` etc.

Both `if` conditions here should be unified into one with `||` op.

Before this patch, only the `~` operator and its combinations didn't work correctly, so it's not a big deal.